make/common/JdkNativeCompilation.gmk

```diff
@@ -109,19 +249,30 @@ JDK_RCFLAGS=$(RCFLAGS) \
 # SetupNativeCompilation. This macro also adds the following additional
 # arguments:
 #
-#   EXTRA_RCFLAGS -- additional RCFLAGS to append.
-#   EXTRA_HEADER_DIRS -- additional directories to look for headers in
-#   EXTRA_SRC -- additional directories to look for source in
+#   SRC -- this is passed on after preprocessing. The preprocessing will pass on
+#     absolute paths unchanged, but relative paths will be expanded to include
+#     all directories with this name in $MODULE/[shared|$OS|$OSTYPE]/native.
+#   EXTRA_SRC -- additional directories to look for source in. This is processed
+#     like SRC.
 #   EXCLUDE_SRC_PATTERNS -- exclude source dirs matching these patterns from
 #     appearing in SRC.
+#   EXTRA_HEADER_DIRS -- additional directories to look for headers in. This can
+#     be specified either as an absolute path, or relative directory names which
+#     are preprocessed like SRC, or in the format <module>:<directory>, which
+#     will be processed like SRC but for the given module. The name "*:libjvm"
+#     is a virtual library that will be replaced with proper values for hotspot.
 #   HEADERS_FROM_SRC -- if false, does not add source dirs automatically as
 #     header include dirs. (Defaults to true.)
-#   SRC -- this is passed on, but preprocessed to accept source dir designations
-#   RC_FILEDESC -- override the default FILEDESC for Windows version.rc
-#     such as "java.base:headers".
-#   JDK_LIBS -- libraries generated by the JDK build system to link against
+#   JDK_LIBS -- libraries generated by the JDK build system to link against.
+#     These take the form <module>:<basename>. If the module is java.base,
+#     the entire java.base: prefix can be omitted. If the module is @, this will
+#     be replaced with the current module. The gtest module is a virtual module
```

Comment:
I understand wanting to optimize for the common case, which is linking against libjava and libjvm in java.base, but I'm wondering if it wouldn't be better to be more coherent with the EXTRA_HEADER_DIRS argument. No prefix there means current module, while no prefix here means java.base. I think we will be tripping over this inconsistency in the future.

I also note that headers are referred to as `libfoo` while libraries are only `foo`. This is harder to do anything about as on references a directory name, while the other is the build library. Still something to consider. It would be neat if these options could take the same syntax when appropriate.